### ISSUE TITLE
Reinvent start page

### ISSUE THREAD

**Issue by tvlooy** [Intracto/SecretSanta#505]
I would like to see a new start page in 2021.

Who's the administrator? [ name ] [ email ]
[v] administrator participates in the party would fix #504 

That's it. No need to add 3 participants, no CSV import. First you get the link to your management page.

On the management page you will no longer have the option to choose the join mode like implemented in #502 but you always have a join link:
- if you add people by CSV (now implemented on the homepage), or manually. These people will get an invite to join with a [confirm] [do not want]
- you can share the join link and people will be able to join from the link by providing their own info

No checkbox "I have permission to use these email addresses". Because we will make it opt-in.

If the party starts only confirmed people will be added in the party. Unconfirmed or rejected will be removed from the list.

Adding / removing people will follow the same principle. Remove does not require a verification from the participant, but a mail will be sent. Adding a person will first have to be confirmed by the person.

Joining by invite link will not be possible after the party started.

**Comment by TomCan** [comment]
Hmmz, I have serious reservations regarding these proposed changes...

- Having to "verify first" creates a barrier, granted a small one. The current homepage is very inviting and basically screams to get started right away. When they first add names and need to verify after, there's an incentive to use our tool instead of finding another one since they already done some of the work by adding the participants. I would rather just add the option to continue without adding participants, rather than removing the option to add them from the homepage. Add some copy to highlight the other options, and maybe combine with the #504 checkbox.

- Having each participant join/opt-in also means that you can't start matching until everyone has in fact joined the party. This might be a problem for less tech-savvy users, or for users that basically expect someone else to make all arrangements and just want to know who to buy for. It creates more steps and more e-mails and requires more management/monitoring for the manager of the party. imho this will impact user experience, both for the participants as for the manager, in a negative way. If would probably solve a very small problem of an occasional unwillingly added participant, but will probably create more and bigger problems instead.

**Comment by memic84** [comment]
What about a Social Login? If we offer two options: social login or get started with email.. With the social login, you won't have the barrier of verifying the user through them opening their mail client.

With one-click with the Social Login, you could start adding all the names on the homepage, without having to open your mail client.

Don't know how this will impact the whole flow and possibilities regarding the current backend architecture.

**Comment by tvlooy** [comment]
adding participants would then also have to be based on social accounts. Ideally you need to be able to mix accounts, eg 10 people from facebook but 5 that don't have facebook and should be added by email ...

it's definitely an option but not a small one.

what social platforms would you want to support?

**Comment by RubenHollevoet** [comment]
> What about a Social Login? If we offer two options: social login or get started with email.. With the social login, you won't have the barrier of verifying the user through them opening their mail client.
> 
> With one-click with the Social Login, you could start adding all the names on the homepage, without having to open your mail client.
> 
> Don't know how this will impact the whole flow and possibilities regarding the current backend architecture.

I also like the idea since it will give the allowance to display all your active and past parties instantly without having to search for the admin links in your mailbox. Reusing parties or managing multiple parties will become much easier.

**Comment by TomCan** [comment]
Just dumping a number of ways for the signup, to be discussed in detail to see which ones (can be multiple) we're going to implement:

**current situation**
Manager creates party and manually adds participants or imports though csv. Manager starts party and e-mails are sent out to all users in party.
pro: manager is not dependant on participants to organise the party. Good for parties that assigned someone to "make it happen".
con: no explicit consent by participants (gdpr), manager declares he has permission

**invite users**
Manager creates party and manually invites participants or imports through csv. Participants are sent an invite e-mail where they need to click/approve that they want to particiate. When manager starts the party, all unconfirmed participants are removed from the party.
pro: explicit consent by participants (gdpr) with exception of invitation e-mail (manager declares he has permission).
con: more e-mails are sent out

**share a link**
Manager creates a party and gets a unique link that can be shared through other channels outside the scope of the site (e-mail, facebook, whatsapp, slack...). Anyone who has the link, can add themselves to the party (with e-mail confirmation).
pro: explicit consent by participants (gdpr), no processing of PI unless explicitly added.
pro: allows parties where larger communities can organise a party without massive administrative burden for manager (might need additional info like shipping info though)
con: still needs e-mail confirmation to join party 
con: users might forget to register and miss out on the party

**add without e-mail**
Manager creates party and manually adds participants or imports though csv. E-mail is optional. Manager starts party and e-mails are sent out to all users in party that have an e-mailaddress associated. Manager get's a list of participants without e-mail and their matches, for manual distribution. 
pro: no processing of e-mailaddresses for those participants
pro: allows granny to participate
con: less interaction with site/tool. No option for wishlist, of sending a message to your match.

**Comment by RubenHollevoet** [comment]
2 more options

**link based parties (email free)**
Manager creates a party. Participants are added by their name, no email.
The manager starts the party and gets a participant link to share with all participants through a channel of choice.
When visiting the participants page you need to pick your name form a list of participants after which his/her wishlist and match is shown. Once a name is picked, the combo username + party hash are saved in browser storage so you only need to pick your name once per party per device and users aren't encouraged to checkout other peoples match and wishlist.
pro: eliminates almost all emails, fully GDPR compliant 
con: sneaky participants can see all matches or ruin someone else their wishlist

**invite users - extended (extended version of @TomCan 's proposal)**
Another con of the invite users proposal above would be that the start of the party is blocked by people who are not confirming their participation invite (ex someone who's actually willing to participate but not in the position of confirming due to illness, holidays, any kind of mail issues, ...)
When starting the party, the manager should have the choice to either remove or keep unconfirmed users. Unconfirmed users will however not receive any more mails until they confirm.
pro: start of the party can't be blocked by unconfirmed participants
pro/con: see proposal of @TomCan

**Comment by Hydrog3n** [comment]
Hello, I jump in the conversation because I was come to open this kind of feature. 

I would add an option the administrator can limit registration by email pattern like slack do (@test.com). Only when people come from the shared link. The invitation by email by pass the limit. To allow external to participate.
